extensions/HuggingFaceTransformers/python/src/aiconfig_extension_hugging_face_transformers/text_2_image.py

```diff
@@ -0,0 +1,341 @@
+import base64
+import copy
+import io
+import itertools
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+from diffusers import AutoPipelineForText2Image
+from diffusers.pipelines.stable_diffusion import StableDiffusionPipelineOutput
+from diffusers.pipelines.stable_diffusion_xl.pipeline_output import StableDiffusionXLPipelineOutput
+from PIL import Image
+from transformers import Pipeline
+
+
+from aiconfig.default_parsers.parameterized_model_parser import ParameterizedModelParser
+from aiconfig.model_parser import InferenceOptions
+from aiconfig.schema import ExecuteResult, Output, Prompt, PromptMetadata
+from aiconfig.util.params import resolve_prompt
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from aiconfig.Config import AIConfigRuntime
+
+
+# Step 1: define Helpers
+def refine_pipeline_creation_params(model_settings: Dict[str, Any]) -> List[Dict[str, Any]]:
+    """
+    Refines the pipeline creation params for the HF text2Image generation api. 
+    Defers unsupported params as completion params, where they can get processed in 
+    `refine_image_completion_params()`. The supported keys were found by looking at
+    the HF text2Image API: 
+    https://huggingface.co/docs/diffusers/v0.24.0/en/api/pipelines/auto_pipeline#diffusers.AutoPipelineForText2Image
+
+    Note that this is not the same as the image completion params, which are passed to 
+    the pipeline later to generate the image: 
+    https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/text2img#diffusers.StableDiffusionPipeline.__call__
+    """
+
+    supported_keys = {
+        "torch_dtype",
+        "force_download",
+        "cache_dir",
+        "resume_download",
+        "proxies",
+        "output_loading_info",
+        "local_files_only",
+        "use_auth_token",
+        "revision",
+        "custom_revision",
+        "mirror",
+        "device_map",
+        "max_memory",
+        "offload_folder",
+        "offload_state_dict",
+        "low_cpu_mem_usage",
+        "use_safetensors",
+        "variant",
+    }
+
+    pipeline_creation_params : Dict[str, Any] = {}
+    completion_params : Dict[str, Any] = {}
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            pipeline_creation_params[key.lower()] = model_settings[key]
+        else:
+            if key.lower() == "kwargs" and isinstance(model_settings[key], Dict):
+                completion_params.update(model_settings[key])
+            else:
+                completion_params[key.lower()] = model_settings[key]
+
+    return [pipeline_creation_params, completion_params]
```

Comment:
why do we need to split it up into two parts? Mainly for myself to understand but trying to understand the logic flow

Reply:
One is a set of params for creating the pipeline generator, while another is a set of params that the pipeline generator then takes in for creating the model itself